### PR TITLE
Allow enabling hostNetwork mode in Helm chart

### LINKED ISSUE #62: Webhook error when creating a bundle
I installed cert-manager-trust via Helm with the 0.2.0 chart (https://artifacthub.io/packages/helm/cert-manager/cert-manager-trust). I had no errors when installing it. When I try to create a bundle resource, I get the following message:

```
Internal error occurred: failed calling webhook "trust.cert-manager.io": Post "https://cert-manager-trust.cert-manager.svc:443/validate?timeout=10s": context deadline exceeded
```

I increased to 10s the timeout to discard that as an issue. When debugging, I saw the following logs in the trust pod:
```
I1018 09:07:55.682760       1 webhook.go:32] trust/webhook "msg"="registering webhook endpoints"
I1018 09:07:55.683436       1 internal.go:362] trust/manager "msg"="Starting server" "addr"={"IP":"::","Port":9402,"Zone":""} "kind"="metrics" "path"="/metrics"
I1018 09:07:55.683575       1 internal.go:362] trust/manager "msg"="Starting server" "addr"={"IP":"::","Port":6060,"Zone":""} "kind"="health probe"
I1018 09:07:55.683872       1 leaderelection.go:248] attempting to acquire leader lease cert-manager/cert-manager-trust-leader-election...
I1018 09:08:00.277660       1 leaderelection.go:258] successfully acquired lease cert-manager/cert-manager-trust-leader-election
I1018 09:08:00.283090       1 controller.go:178] trust/manager/controller/bundle "msg"="Starting EventSource" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "source"="kind source: *v1alpha1.Bundle"
I1018 09:08:00.283556       1 controller.go:178] trust/manager/controller/bundle "msg"="Starting EventSource" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "source"="kind source: *v1.PartialObjectMetadata"
I1018 09:08:00.284286       1 controller.go:178] trust/manager/controller/bundle "msg"="Starting EventSource" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "source"="kind source: *v1.Namespace"
I1018 09:08:00.284315       1 controller.go:178] trust/manager/controller/bundle "msg"="Starting EventSource" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "source"="kind source: *v1.PartialObjectMetadata"
I1018 09:08:00.284335       1 controller.go:178] trust/manager/controller/bundle "msg"="Starting EventSource" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "source"="kind source: *v1.PartialObjectMetadata"
I1018 09:08:00.284355       1 controller.go:186] trust/manager/controller/bundle "msg"="Starting Controller" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle"
I1018 09:08:00.283375       1 recorder.go:103] trust/manager/events "msg"="Normal"  "message"="cert-manager-trust-6c4bfdb98f-9b6zm_27895cf5-9a41-472b-b3b0-39252ad0a2af became leader" "object"={"kind":"ConfigMap","namespace":"cert-manager","name":"cert-manager-trust-leader-election","uid":"34490034-b485-42b3-8d5b-d5e0f05715ba","apiVersion":"v1","resourceVersion":"25487825"} "reason"="LeaderElection"
I1018 09:08:00.284555       1 recorder.go:103] trust/manager/events "msg"="Normal"  "message"="cert-manager-trust-6c4bfdb98f-9b6zm_27895cf5-9a41-472b-b3b0-39252ad0a2af became leader" "object"={"kind":"Lease","namespace":"cert-manager","name":"cert-manager-trust-leader-election","uid":"22f79707-6aca-497c-92ef-e964990795c6","apiVersion":"coordination.k8s.io/v1","resourceVersion":"25487826"} "reason"="LeaderElection"
I1018 09:08:00.385279       1 controller.go:220] trust/manager/controller/bundle "msg"="Starting workers" "reconciler group"="trust.cert-manager.io" "reconciler kind"="Bundle" "worker count"=1
```
I tried doing a port-forward to the service and I had issues with the certificate. I get a `404 page not found` error. In the pod logs I can find:
```
2022/10/18 09:25:41 http: TLS handshake error from 127.0.0.1:53246: remote error: tls: unknown certificate
```
Then if I do the port-forward to the pod itself, no SSL is needed. And if I go to /readyz, it answers `ok`. So I am suspecting there is an issue with the service forcing to use SSL, and maybe not being able to connect because the certificate is invalid.

### PR DESCRIPTION
**Pull request by cablespaghetti**
Taking inspiration from the cert-manager helm chart, this fixes the prematurely closed #62 by making it possible to start the trust manager pods in hostNetwork mode. I've tested this on my cluster and it fixes issues with EKS running with a non-standard CNI.

### CODE REVIEW

**Comment by ChevronTango** [comment]
I just raised https://github.com/cert-manager/trust-manager/issues/159 as we've been encountering this issue. Glad to see that there's already a fix in the works. It's a simple enough change but we'd be very ken to see this merged and released ASAP.

**Comment by J1235** [comment]
We're using the fairly standard Calico CNI with EKS and this PR would solve the problems we see when doing updates, as confirmed by a patch we've had to do after deployment.

**Review by erikgb — CHANGES_REQUESTED** [review]
/ok-to-test

**Comment by erikgb** [comment]
@cablespaghetti Thanks for this. "Please run 'make -s update-helm-docs" to fix the build.

**Comment by cablespaghetti** [comment]
Sorry for the delay. I've sorted this now.

**Comment by erikgb** [comment]
@cablespaghetti It seems like the docs marker for the new Helm value is not properly formatted. Could you fix this and regenerate the docs - to ensure the new value is documented in the Helm README?

**Review by erikgb — APPROVED** [review]
/lgtm

**Comment by J1235** [comment]
Nearly there?

File: deploy/charts/trust-manager/README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Kubernetes: `>= 1.25.0`
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | affinity | object | `{}` | Kubernetes Affinty; see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#affinity-v1-core |
+| app.hostNetwork | bool | `false` | Specifies if the app should be started in hostNetwork mode. Required for use in some managed kubernetes clusters (such as AWS EKS) with custom CNI. |
```
Comment (inteon) [inline deploy/charts/trust-manager/README.md:28]: I think this should be a webhook option (so `app.webhook.hostNetwork`)

Comment (inteon) [reply deploy/charts/trust-manager/README.md:28]: Also, I think we want to set `dnsPolicy: ClusterFirstWithHostNet` when hostNetwork is enabled (based on what we have in cert-manager): https://github.com/cert-manager/cert-manager/blob/ef3b66aebafa8467ecceb2e8567b4d62b400b2bb/deploy/charts/cert-manager/templates/webhook-deployment.yaml#L58-L60.

**Comment by cablespaghetti** [comment]
I think this is good to go now.

**Comment by inteon** [comment]
Thanks a lot for this contribution!

**Pull request merged**